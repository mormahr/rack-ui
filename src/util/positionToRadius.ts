export function positionToRadius(
  angleDegrees: number,
  radius: number,
  cx: number,
  cy: number = cx,
  base: number = 90,
): [number, number] {
  const angle = ((angleDegrees - base) * Math.PI) / 180.0

  return [cx + radius * Math.cos(angle), cy + radius * Math.sin(angle)]
}
