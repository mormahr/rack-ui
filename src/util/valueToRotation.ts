export function valueToRotation(value: number, openingAngle: number) {
  return (value - 0.5) * 2 * openingAngle
}
