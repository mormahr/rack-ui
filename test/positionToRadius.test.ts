import { positionToRadius } from "../src/util/positionToRadius"

describe("positionToRadius", () => {
  it("default 15° middle", () => {
    expect(positionToRadius(15, 50, 50)).toMatchSnapshot()
  })

  it("default 15°", () => {
    expect(positionToRadius(15, 50, 50, 50)).toMatchSnapshot()
  })

  it("default 30°", () => {
    expect(positionToRadius(30, 50, 50, 50)).toMatchSnapshot()
  })

  it("default 30°, translated", () => {
    expect(positionToRadius(30, 50, 40, 60)).toMatchSnapshot()
  })

  it("default 30°, translated, different radius", () => {
    expect(positionToRadius(30, 20, 40, 60)).toMatchSnapshot()
  })

  it("different base, 30°", () => {
    expect(positionToRadius(30, 20, 40, 60, 180)).toMatchSnapshot()
  })
})
