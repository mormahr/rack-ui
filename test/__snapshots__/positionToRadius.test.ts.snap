// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`positionToRadius default 15° 1`] = `
Array [
  62.940952255126035,
  1.7037086855465873,
]
`;

exports[`positionToRadius default 15° middle 1`] = `
Array [
  62.940952255126035,
  1.7037086855465873,
]
`;

exports[`positionToRadius default 30° 1`] = `
Array [
  75,
  6.698729810778069,
]
`;

exports[`positionToRadius default 30°, translated 1`] = `
Array [
  65,
  16.69872981077807,
]
`;

exports[`positionToRadius default 30°, translated, different radius 1`] = `
Array [
  50,
  42.67949192431123,
]
`;

exports[`positionToRadius different base, 30° 1`] = `
Array [
  22.679491924311225,
  50,
]
`;
